### ISSUE TITLE
Point-NeRF: Point-based Neural Radiance Fields

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#170]
## TL; DR

- Combine point cloud representation obtained from (relatively) classical MVS with NeRF volume rendering to achieve faster-to-train and generalizable implicit neural representations.

## Motivation

- NeRF variants takes painfully long time to train, and they are overfitted to each individual scene, lacking the capacity to generalize.

## Formulation

![image](https://user-images.githubusercontent.com/46308407/163399809-48e6063b-6828-43a3-9a60-2b2bb0ef6462.png)

### Point Cloud Representation

- Leverage off-the-shelf MVS network (*or COLMAP works just fine*) to infer point clouds from a set of posed images.
- Employ additional image encoder to assign *features* to each point clouds.
- This way, the framework can more or less generalize to multiple scenes and adapt quickly to new scene composition.

### Fine-tuning

- With initial point cloud representations, coordinate-MLP based NeRF model is trained to output fine radiance and density.
- Features from neighboring points (*in kNN manner*) are aggregated via a shallow MLP for better local representation.
- Fine-tuning the NeRF model takes incredibly short amount of time as the model is given an initial coarse point cloud representation (*so it doesn't have to sample and evaluate points in the void*).

![image](https://user-images.githubusercontent.com/46308407/163400656-21a018a4-ef40-4ea2-9631-ec7c5bf90da9.png)

### Point Pruning / Growing

- Some point cloud estimates can be noisy due to imperfection of the MVS process.
- Discard those with low *confidence* at regular intervals.
- Also, add points where the density value is predicted high. (Note that the 3D scene is first voxelized, and those that do not contain point clouds are deemed *empty*. We shoot rays and sample points along only in the *occupied* voxels. Some of the points could display high density estimate while there is no corresponding point cloud nearby. In that case, we *grow* the point cloud representation by adding such point)

## Evaluation

![image](https://user-images.githubusercontent.com/46308407/163401284-13857d2e-389f-4778-b9cb-3a068d10dfbb.png)
![image](https://user-images.githubusercontent.com/46308407/163401296-990a695a-e23e-4af9-8658-cac2821aafe0.png)
![image](https://user-images.githubusercontent.com/46308407/163401309-d2c7f36b-6e24-4de0-ba3d-7a0b7388b0f8.png)
![image](https://user-images.githubusercontent.com/46308407/163401332-2021bb42-e448-4b7a-90b4-452e117ab829.png)
![image](https://user-images.githubusercontent.com/46308407/163401350-b7cd8a8d-705c-4b50-9fc4-88d2345383fd.png)
![image](https://user-images.githubusercontent.com/46308407/163401364-cd398645-3d4b-4d4f-bf47-a982a930f3aa.png)

## Comments

- Smart idea to connect classical point cloud representations with NeRF volume rendering.
- Somewhat similar intuitions to NSVF, with less of a *wow factor* in terms of methodology (*personal opinion*) but impressive performance.
- It's always wise to study the classics, as we never know when they would rise above the coffin.
